signerv2/signer.go

```diff
@@ -0,0 +1,81 @@
+package signerv2
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"errors"
+	sdkEcdsa "github.com/Layr-Labs/eigensdk-go/crypto/ecdsa"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"math/big"
+)
+
+type SignerFn func(ctx context.Context, address common.Address) (bind.SignerFn, error)
+
+// SignerFactory creates a SignerFn that is bound to a specific ChainID
+type SignerFactory func(chainID *big.Int) SignerFn
```

Comment:
This interface is suuuuper confusing. We're creating a SignerFactory, that creates a SignerFn, that creates a bind.SignerFn??? Why do we need all this complexity?

Reply:
updated